src/backend/distributed/commands/truncate.c

```diff
@@ -267,13 +267,13 @@ ErrorIfUnsupportedTruncateStmt(TruncateStmt *truncateStatement)
 
 		ErrorIfIllegallyChangingKnownShard(relationId);
 
-		if (IsCitusTable(relationId) && IsForeignTable(relationId))
+		if (IsForeignTable(relationId) &&
+			IsCitusTableType(relationId, CITUS_LOCAL_TABLE) &&
+			!IsCoordinator())
 		{
 			ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-							errmsg("truncating distributed foreign tables is "
-								   "currently unsupported"),
-							errhint("Consider undistributing table before TRUNCATE, "
-									"and then distribute or add to metadata again")));
+							errmsg("truncating tables that are added to metadata can "
```

Comment:
minor: `truncating foreign tables...` otherwise we'd be giving misleading message to the users

Also, `done` -> `executed`